src/main.py

```diff
@@ -1,2 +1,39 @@
+from rdkit import Chem
+
+
 def substructure_search(mols, mol):
-    pass
\ No newline at end of file
+    """
+      Func returns a list of all molecules from argument 1 that contain substructure from argument 2.
+
+      Arguments:
+          mols (list): A list of SMILES strings containing the molecules to search.
+          mol (str): A SMILES string which has the substructure to search for.
+
+      Returns:
+          list: A list of SMILES strings representing the molecules that contain the substructure.
+          str: An error message if an exception occurs.
+      """
+    try:
+        substructure = Chem.MolFromSmiles(mol)
+        if substructure is None:
+            raise ValueError("Unknown substructure SMILES string")
+
+        matches = []
+        for mol_smiles in mols:
+            try:
+                molecule = Chem.MolFromSmiles(mol_smiles)
+                if molecule is None:
+                    raise ValueError(f"Invalid molecule SMILES string: {mol_smiles}")
+
+                if molecule.HasSubstructMatch(substructure):
+                    matches.append(mol_smiles)
+            except ValueError as ve:
+                print(ve)
```

Comment:
> It's bad practice to stifle exceptions; this can lead to big errors later on.

Thank you for review, what would be the more optimal way to utilise it properly?